broker/broker.go

```diff
@@ -16,18 +16,25 @@ import (
 	"github.com/pivotal-cf/brokerapi/domain"
 )
 
+var (
+	ErrNilResponse                  = fmt.Errorf("invalid nil response from provider")
+	ErrNotImplemented               = fmt.Errorf("provider does not implement this operation")
+	ErrAsyncProvisionNotImplemented = fmt.Errorf("provider returned async response but does not implement AsyncProvisioner")
+	ErrAsyncBindNotImplemented      = fmt.Errorf("provider returned async response but does not implement AsyncBinder")
+)
+
 const (
 	locketMaxTTL = 30
 )
 
 type Broker struct {
 	config       Config
-	Provider     provider.ServiceProvider
+	Provider     provider.Provider
 	logger       lager.Logger
 	LocketClient locket_models.LocketClient
 }
 
```

Comment:
```suggestion
var _ domain.ServiceBroker = &Broker{}

```

It wasn't clear to me on first reading why this code had the parameters it did (for example, why does `Services()` take a context, and then not use it?).  This line would a) document that intention, and b) get the compiler to enforce that a *Broker is a domain.ServiceBroker.